File: app/src/main/java/com/ptvon/domain/model/RouteType.kt
package com.ptvon.domain.model

/**
 * PTV `route_type` identifiers, paired with their canonical brand colours
 * (used by the PID-style UI). Colours are stored as ARGB longs so the domain
 * layer stays Compose-free; the UI maps them to `Color(...)`.
 */
enum class RouteType(val apiValue: Int, val displayName: String, val brandColor: Long) {
    TRAIN(0, "Train", 0xFF3F7BFF),    // friendly blue
    TRAM(1, "Tram", 0xFF35C07A),      // friendly green
    BUS(2, "Bus", 0xFFFF9D4D),        // friendly orange
    VLINE(3, "V/Line", 0xFFA06CFF),   // friendly purple
    NIGHT_BUS(4, "Night Bus", 0xFFFF9D4D);

    companion object {
        fun fromApiValue(value: Int): RouteType =
            entries.firstOrNull { it.apiValue == value } ?: BUS
    }
}

File: ios/Shared/RouteType.swift
import SwiftUI

/// PTV transport modes (route_type) with friendly brand colours + SF Symbols.
enum RouteType: Int, CaseIterable, Codable {
    case train = 0
    case tram = 1
    case bus = 2
    case vline = 3
    case nightBus = 4

    var displayName: String {
        switch self {
        case .train: return "Train"
        case .tram: return "Tram"
        case .bus: return "Bus"
        case .vline: return "V/Line"
        case .nightBus: return "Night Bus"
        }
    }

    var colorHex: String {
        switch self {
        case .train: return "3F7BFF"
        case .tram: return "35C07A"
        case .bus, .nightBus: return "FF9D4D"
        case .vline: return "A06CFF"
        }
    }

    var symbol: String {
        switch self {
        case .train, .vline: return "train.side.front.car"
        case .tram: return "tram.fill"
        case .bus, .nightBus: return "bus.fill"
        }
    }

    var color: Color { Color(hex: colorHex) }

    static func from(_ value: Int) -> RouteType { RouteType(rawValue: value) ?? .bus }
}
